### ISSUE TITLE
[Bug]: Move or copy operation failed on Nextcloud server 29.0.6

### ISSUE THREAD

**Issue by gillesF78** [nextcloud/server#47713]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

Moving a folder shared with me fails as the screen capture below shows:

This shared folder can be renamed by me successfully.

Here follows some reports and screen capture:
![NC_MoveOrCopyActionFailed](https://github.com/user-attachments/assets/04503c9f-a400-4a83-8cbf-8cdf590107f8)

### Steps to reproduce

1. Ask a user to share a folder with you
2. Try to move the shared folder to a subfolder you own
3. Error message "Move or copy operation failed" is shown

### Expected behavior

The shared folder should be moved to your selected subfolder.

### Nextcloud Server version

29

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.1

### Web server

Apache (supported)

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

Upgraded to a MAJOR version (ex. 28 to 29)

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
<--! Paste this in GitHub report -->

<details>
<summary>NextcloudPi diagnostics</summary>



NextcloudPi version  v1.54.3
NextcloudPi image    NextCloudPi_11-27-20
OS                   Debian GNU/Linux 12. 6.1.21-v8+ (aarch64)
automount            no
USB devices          sda sdb sdc
datadir              /media/hd3/data
data in SD           no
data filesystem      ext2/ext3
data disk usage      832G/1,8T
rootfs usage         14G/219G
swapfile             /var/swap
dbdir                /var/lib/mysql
Nextcloud check      ok
Nextcloud version    29.0.5.1
HTTPD service        up
PHP service          up
MariaDB service      up
Redis service        up
HPB service          up
Postfix service      down
Internet check       ok
Public IPv4          ***REMOVED SENSITIVE VALUE***
Public IPv6          ***REMOVED SENSITIVE VALUE***
Port 80              open
Port 443             open
IP                   ***REMOVED SENSITIVE VALUE***
Gateway              ***REMOVED SENSITIVE VALUE***
Interface            eth0
Certificates         ***REMOVED SENSITIVE VALUE***
NAT loopback         no
Uptime               17:29

```
</details>

<details>
<summary>Nextcloud configuration</summary>

```

{
    "system": {
        "config_is_read_only": true,
        "default_phone_region": "FR",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "localhost",
             ***REMOVED SENSITIVE VALUE***
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "29.0.6.1",
        "overwrite.cli.url": " ***REMOVED SENSITIVE VALUE***",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "filelocking.enabled": true,
        "memcache.local": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 0,
            "timeout": 0
        },
        "tempdirectory": "\/media\/hd3\/data\/tmp",
        "preview_max_x": "2048",
        "preview_max_y": "2048",
        "preview_max_memory": "4096",
        "preview_max_filesize_image": "256",
        "jpeg_quality": "60",
        "forcessl": true,
        "overwriteprotocol": "https",
        "auth.bruteforce.protection.enabled": false,
        "csrf.optout": [
            "\/^WebDAVFS\/",
            "\/^Microsoft-WebDAV-MiniRedir\/"
        ],
        "maintenance": false,
        "theme": "",
        "mail_sendmailmode": "smtp",
        "mail_smtpsecure": "ssl",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "465",
        "mail_smtpmode": "smtp",
        "mail_smtpauthtype": "LOGIN",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "updater.release.channel": "stable",
        "trusted_proxies": "***REMOVED SENSITIVE VALUE***",
        "debug": false,
        "logfile": "\/media\/hd3\/data\/nextcloud.log",
        "loglevel": 2,
        "data-fingerprint": "396f49904c922a82d5e2d5b5596dae8f",
        "maintenance_window_start": 1,
        "app_install_overwrite": [
            "nextcloudpi",
            "previewgenerator"
        ]
    }
}

```
</details>

<details>
<summary>HTTPd logs</summary>

```

[Tue Sep 03 04:33:10.030291 2024] [authz_core:error] [pid 58001:tid 58001] [client 46.101.143.135:28479] AH01630: client denied by server configuration: /var/www/ncp-web/
[Tue Sep 03 04:33:10.448429 2024] [authz_host:error] [pid 57995:tid 57995] [client 164.92.161.142:26170] AH01753: access check of 'localhost' to / failed, reason: unable to get the remote host name
[Tue Sep 03 04:33:10.448602 2024] [authz_core:error] [pid 57995:tid 57995] [client 164.92.161.142:26170] AH01630: client denied by server configuration: /var/www/ncp-web/
[Tue Sep 03 04:33:10.452019 2024] [authz_host:error] [pid 61452:tid 61452] [client 164.90.212.153:48774] AH01753: access check of 'localhost' to / failed, reason: unable to get the remote host name
[Tue Sep 03 04:33:10.452114 2024] [authz_core:error] [pid 61452:tid 61452] [client 164.90.212.153:48774] AH01630: client denied by server configuration: /var/www/ncp-web/
[Tue Sep 03 04:33:11.152620 2024] [authz_core:error] [pid 58248:tid 58248] [client 52.76.71.100:29784] AH01630: client denied by server configuration: /var/www/ncp-web/favicon.ico
[Tue Sep 03 08:31:04.977220 2024] [authz_core:error] [pid 76684:tid 76684] [client 64.62.197.6:12635] AH01630: client denied by server configuration: /var/www/ncp-web/
[Tue Sep 03 08:31:27.294070 2024] [authz_core:error] [pid 74658:tid 74658] [client 64.62.197.16:20111] AH01630: client denied by server configuration: /var/www/ncp-web/
[Tue Sep 03 09:29:46.961409 2024] [authz_host:error] [pid 80109:tid 80109] [client 205.210.31.72:60814] AH01753: access check of 'localhost' to / failed, reason: unable to get the remote host name
[Tue Sep 03 09:29:46.961561 2024] [authz_core:error] [pid 80109:tid 80109] [client 205.210.31.72:60814] AH01630: client denied by server configuration: /var/www/ncp-web/
[Tue Sep 03 09:50:11.742197 2024] [authz_host:error] [pid 80191:tid 80191] [client 205.210.31.27:64428] AH01753: access check of 'localhost' to / failed, reason: unable to get the remote host name
[Tue Sep 03 09:50:11.742408 2024] [authz_core:error] [pid 80191:tid 80191] [client 205.210.31.27:64428] AH01630: client denied by server configuration: /var/www/ncp-web/
[Tue Sep 03 13:03:37.284152 2024] [core:error] [pid 80191:tid 80191] [client 8.222.175.173:40094] AH10244: invalid URI path (/cgi-bin/.%2e/.%2e/.%2e/.%2e/.%2e/.%2e/.%2e/.%2e/.%2e/.%2e/bin/sh)
[Tue Sep 03 13:14:18.392953 2024] [authz_host:error] [pid 80127:tid 80127] [client 206.168.34.40:54918] AH01753: access check of 'localhost' to / failed, reason: unable to get the remote host name
[Tue Sep 03 13:14:18.393163 2024] [authz_core:error] [pid 80127:tid 80127] [client 206.168.34.40:54918] AH01630: client denied by server configuration: /var/www/ncp-web/
[Tue Sep 03 13:14:27.167374 2024] [authz_host:error] [pid 84914:tid 84914] [client 206.168.34.40:59590] AH01753: access check of 'localhost' to / failed, reason: unable to get the remote host name
[Tue Sep 03 13:14:27.167461 2024] [authz_core:error] [pid 84914:tid 84914] [client 206.168.34.40:59590] AH01630: client denied by server configuration: /var/www/ncp-web/
[Tue Sep 03 15:29:55.637131 2024] [core:error] [pid 88149:tid 88149] [client 147.45.76.168:40996] AH10244: invalid URI path (/cgi-bin/.%2e/.%2e/.%2e/.%2e/.%2e/.%2e/.%2e/.%2e/.%2e/.%2e/bin/sh)
[Tue Sep 03 15:29:56.105831 2024] [core:error] [pid 88157:tid 88157] [client 147.45.76.168:41012] AH10244: invalid URI path (/cgi-bin/%%32%65%%32%65/%%32%65%%32%65/%%32%65%%32%65/%%32%65%%32%65/%%32%65%%32%65/%%32%65%%32%65/%%32%65%%32%65/bin/sh)
[Tue Sep 03 15:30:46.113198 2024] [core:error] [pid 91676:tid 91676] [client 47.250.82.130:51386] AH10244: invalid URI path (/cgi-bin/.%2e/.%2e/.%2e/.%2e/.%2e/.%2e/.%2e/.%2e/.%2e/.%2e/bin/sh)

```
</details>
```


### List of activated Apps

```shell
Enabled:
  - activity: 2.21.1
  - audioplayer: 3.4.1
  - bruteforcesettings: 2.9.0
  - calendar: 4.7.16
  - circles: 29.0.0-dev
  - cloud_federation_api: 1.12.0
  - comments: 1.19.0
  - contacts: 6.0.0
  - contactsinteraction: 1.10.0
  - dashboard: 7.9.0
  - dav: 1.30.1
  - federatedfilesharing: 1.19.0
  - files: 2.1.1
  - files_downloadlimit: 2.0.0
  - files_pdfviewer: 2.10.0
  - files_reminders: 1.2.0
  - files_sharing: 1.21.0
  - files_trashbin: 1.19.0
  - files_versions: 1.22.0
  - logreader: 2.14.0
  - lookup_server_connector: 1.17.0
  - maps: 1.4.0
  - metadata: 0.20.0
  - notes: 4.10.1
  - notifications: 2.17.0
  - notify_push: 0.7.0
  - oauth2: 1.17.0
  - password_policy: 1.19.0
  - previewgenerator: 5.6.0
  - privacy: 1.13.0
  - provisioning_api: 1.19.0
  - recommendations: 2.1.0
  - related_resources: 1.4.0
  - serverinfo: 1.19.0
  - settings: 1.12.0
  - sharebymail: 1.19.0
  - support: 1.12.0
  - systemtags: 1.19.0
  - tasks: 0.16.1
  - text: 3.10.1
  - theming: 2.4.0
  - twofactor_backupcodes: 1.18.0
  - viewer: 2.3.0
  - workflowengine: 2.11.0
Disabled:
  - admin_audit: 1.19.0
  - encryption: 2.17.0
  - federation: 1.19.0 (installed 1.10.1)
  - files_external: 1.21.0 (installed 1.17.0)
  - firstrunwizard: 2.18.0 (installed 2.9.0)
  - nextcloud_announcements: 1.18.0 (installed 1.9.0)
  - nextcloudpi: 0.0.2 (installed 0.0.2)
  - photos: 2.5.0 (installed 1.2.1)
  - survey_client: 1.17.0 (installed 1.17.0)
  - suspicious_login: 7.0.0
  - twofactor_totp: 11.0.0-dev
  - updatenotification: 1.19.1 (installed 1.11.0)
  - user_ldap: 1.20.0
  - user_status: 1.9.0 (installed 1.9.0)
  - weather_status: 1.9.0 (installed 1.4.0)
```

### Nextcloud Signing status

```shell
No errors have been found.
```

### Nextcloud Logs

```shell
There is no line at the "Warning" or "Error" levels.
```

**Comment by kesselb** [comment]
Thank you for reporting this issue  :+1: 

I will need your browser console log to investigate this issue.  Open your console, reload your page and/or do the action leading to this issue and copy/paste the log in this thread.

<details>
<summary>How to access your browser console (Click to expand)</summary>

# Chrome
- Press either CTRL + SHIFT + J to open the “console” tab of the Developer Tools.
- Alternative method:
    1. Press either CTRL + SHIFT + I or F12 to open the Developer Tools.
    2. Click the “console” tab.

# Safari
- Press CMD + ALT + I to open the Web Inspector.
- See Chrome’s step 2. (Chrome and Safari have pretty much identical dev tools.)

# Firefox
- Press CTRL + SHIFT + K to open the Web console (COMMAND + SHIFT + K on Macs).
- or, if Firebug is installed (recommended):
    1. Press F12 to open Firebug.
    2. Click on the “console” tab.
</details>

**Comment by gillesF78** [comment]
xhr-request.js:79 
        
        
        MOVE https://myNextCloud.com/remote.php/dav/files/gilles/20240825_Fete%20La%20Rosiere%20a%20Vinay 403 (Forbidden)
r.10907.window.fetch @ xhr-request.js:79
(anonymes) @ index.mjs:830
(anonymes) @ index.js:2
execute @ index.js:2
patchInline @ index.js:2
(anonymes) @ index.js:2
(anonymes) @ index.js:2
execute @ index.js:2
patchInline @ index.js:2
ee @ index.js:2
(anonymes) @ index.js:2
push.31346.t @ index.js:2
(anonymes) @ index.js:2
(anonymes) @ index.js:2
moveFile @ index.js:2
(anonymes) @ moveOrCopyAction.ts:140
await in (anonymes)
(anonymes) @ index.js:111
M @ index.js:285
(anonymes) @ index.js:135
add @ index.js:99
K @ moveOrCopyAction.ts:89
exec @ moveOrCopyAction.ts:278
await in exec
onActionClick @ FileEntryActions.vue:161
click @ FileEntryActions.vue:4
cn @ vue.runtime.esm.js:3033
n @ vue.runtime.esm.js:1832
cn @ vue.runtime.esm.js:3033
(anonymes) @ vue.runtime.esm.js:3732
onClick @ actionText-fFcUPi2g.mjs:77
handleClick @ NcActionButton.mjs:131
cn @ vue.runtime.esm.js:3033
n @ vue.runtime.esm.js:1832
(anonymes) @ vue.runtime.esm.js:7500
Afficher 27 plus de cadres
Afficher moins

**Comment by solracsf** [comment]
Just to be sure, does this also happens on v29.0**.6** ?

**Comment by susnux** [comment]
There should be a log entry in the Nextcloud log, maybe activate `loglevel => 1` and look for the *MOVE*

**Comment by gillesF78** [comment]
I tried again after increasing the loglevel to 1, and this did not raise any Warning:
![nc-loglevel](https://github.com/user-attachments/assets/7cd08199-0106-40f3-ba9c-51137236f57a)

**Comment by susnux** [comment]
> I tried again after increasing the loglevel to 1, and this did not raise any Warning:

Yes log level would be info or something else, could you please provide us the log from the time range you tried move?

**Comment by gillesF78** [comment]
The "Move or Copy" action does not produce any information in the log file.

**Comment by gillesF78** [comment]
There is a Debug level message in the log:

{"reqId":"ZtdRpu1xjLOs-E1QXkPJgAAAAAo","level":0,"time":"2024-09-03T18:12:54+00:00","remoteAddr":"192.168.0.254","user":"gilles","app":"files","method":"MOVE","url":"/remote.php/dav/files/gilles/20240825_Fete%20La%20Rosiere%20a%20Vinay","message":"It is not allowed to move one mount point into a shared folder","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/128.0.0.0 Safari/537.36","version":"29.0.6.1","data":{"app":"files"},"id":"66d751dd5725c"}

**Comment by gillesF78** [comment]
Moving the share to a new folder succeeds. It seems that only the action move to "Photos de maman" destination folder fails.

There is many "Exception dirty table reads" in the log at the Debug level.

**Comment by gillesF78** [comment]
- when I create a new foler named, I can move the share inside "/New Folder"
- when I create a new folder inside "Photos de maman", I can move the share inside "/Photos de maman/New Folder"

**Comment by solracsf** [comment]
Error comes from:

https://github.com/nextcloud/server/blob/f1706df3676574dd63e7b34aff50d3e4fe136377/lib/private/Files/View.php#L1771-L1797

Is `/Photos de maman` somehow shared (by link, email, group, circle...)?

**Comment by gillesF78** [comment]
> Is `/Photos de maman` somehow shared (by link, email, group, circle...)?

Nope. The destination folder "/Photos de maman" is not shared, as depicted bellow:
![Capture d'écran 2024-09-04 164017](https://github.com/user-attachments/assets/3fd9b97d-516f-4544-ab7d-74cf93db7923)

**Comment by susnux** [comment]
> Is `/Photos de maman` somehow shared (by link, email, group, circle...)?

No but the test is bad, it uses a string compare :see_no_evil: 
`/Photos de maman` is not shared but `/Photos` and `/Photos de maman` starts with `/Photos`....

**Comment by solracsf** [comment]
@gillesF78 are you able to apply these changes to your [lib/private/Files/View.php](https://github.com/nextcloud/server/pull/47756/files#diff-64a02ee1149d33d0747360694c2abf4ef1bdbfc379b4bb40f76c22b5b1fc06d9) file, reload your PHP server, and retry?

**Comment by gillesF78** [comment]
I applied the fix to lib/private/Files/View.php, and it works. "Move or copy" action executed successfully 👍

**Comment by gillesF78** [comment]
Hello

I updated my NC server to version 30.0.0, and unfortunately the fix was not applied.


<details>
<summary>NextcloudPi diagnostics</summary>

```

NextcloudPi version  v1.55.2
NextcloudPi image    NextCloudPi_11-27-20
OS                   Debian GNU/Linux 12. 6.1.21-v8+ (aarch64)
automount            no
USB devices          sda sdb sdc
datadir              /media/hd3/data
data in SD           no
data filesystem      ext2/ext3
data disk usage      841G/1,8T
rootfs usage         13G/219G
swapfile             /var/swap
dbdir                /var/lib/mysql
Nextcloud check      ok
Nextcloud version    30.0.0.14
HTTPD service        up
PHP service          up
MariaDB service      up
Redis service        up
HPB service          up
Postfix service      down
Internet check       ok
Public IPv4          ***REMOVED SENSITIVE VALUE***
Public IPv6          ***REMOVED SENSITIVE VALUE***
Port 80              open
Port 443             open
IP                   ***REMOVED SENSITIVE VALUE***
Gateway              ***REMOVED SENSITIVE VALUE***
Interface            eth0
Certificates         ***REMOVED SENSITIVE VALUE***
NAT loopback         no
Uptime               27days

```
</details>

**Comment by susnux** [comment]
> I updated my NC server to version 30.0.0, and unfortunately the fix was not applied.

As you can see when clicking on the pull request: This is fixed with 30.0.1
